tests/API/Helpers/TokenTest.php

```diff
@@ -4,100 +4,434 @@
+
+    /**
+     * Default Client ID.
+     */
+    const CLIENT_ID = '__test_client_id__';
+
+    /**
+     * Default Client Secret.
+     */
+    const CLIENT_SECRET = '__test_client_secret__';
+
+    /**
+     * Test legacy misspelling.
+     *
+     * @return void
+     */
+    public function testThatSuportedAlgsThrowsException()
     {
-        $client_id     = 'client_id_1';
-        $client_secret = 'client_secret_1';
+        $caught_exception = false;
+        try {
+            new JWTVerifier( [ 'suported_algs' => uniqid() ] );
+        } catch (CoreException $e) {
+            $caught_exception = $this->errorHasString( $e, '`suported_algs` was properly renamed to `supported_algs`' );
+        }
 
-        $generator = new TokenGenerator([ 'client_id' => $client_id, 'client_secret' => $client_secret]);
+        $this->assertTrue($caught_exception);
+    }
 
-        $jwt = $generator->generate(
-            [
-                'users' => [
-                    'actions' => ['read']
-                ]
-            ]
-        );
+    /**
+     * Test for audience config param.
+     *
+     * @return void
+     */
+    public function testThatMissingAudThrowsException()
+    {
+        $caught_exception = false;
+        try {
+            new JWTVerifier([]);
+        } catch (CoreException $e) {
+            $caught_exception = $this->errorHasString( $e, 'The audience is mandatory' );
+        }
 
-        $verifier = new JWTVerifier(
-            [
-                'valid_audiences' => [$client_id],
-                'client_secret' => $client_secret
-            ]
-        );
+        $this->assertTrue($caught_exception);
+    }
+
+    /**
+     * Test unsupported algorithms in the config param.
+     *
+     * @return void
+     */
+    public function testThatOtherAlgsThrowsException()
+    {
+        $caught_exception = false;
+        try {
+            new JWTVerifier( [
+                'valid_audiences' => [ uniqid() ],
+                'supported_algs' => [ 'RS256', 'RS512' ],
+            ] );
+        } catch (CoreException $e) {
+            $caught_exception = $this->errorHasString( $e, 'Cannot support the following algorithm(s): RS512' );
+        }
+
+        $this->assertTrue($caught_exception);
+    }
+
+    /**
+     * Test for missing issuer for RS256 tokens.
+     *
+     * @return void
+     */
+    public function testThatMissingIssThrowsException()
+    {
+        $caught_exception = false;
+        try {
+            new JWTVerifier( [
+                'valid_audiences' => [ uniqid() ],
+                'supported_algs' => [ 'RS256' ],
+            ] );
+        } catch (CoreException $e) {
+            $caught_exception = $this->errorHasString( $e, 'The iss is required' );
+        }
+
+        $this->assertTrue($caught_exception);
+    }
+
+    /**
+     * Test that a secret is required for an HS256 token.
+     *
+     * @return void
+     */
+    public function testThatMissingSecretThrowsException()
```

Comment:
in general, use the method name to describe the test as most as possible. e.g. `testThatMissingSecretWhenUsingHS256ThrowsException`